src/map/itemdb.c

```diff
@@ -2373,13 +2486,15 @@ void do_final_itemdb(void) {
 	itemdb->clear(true);
 
 	itemdb->other->destroy(itemdb->other, itemdb->final_sub);
+	itemdb->options->destroy(itemdb->options, itemdb->options_final_sub);
 	itemdb->destroy_item_data(&itemdb->dummy, 0);
 	db_destroy(itemdb->names);
 }
 
 void do_init_itemdb(bool minimal) {
 	memset(itemdb->array, 0, sizeof(itemdb->array));
 	itemdb->other = idb_alloc(DB_OPT_BASE);
+	itemdb->options = idb_alloc(DB_OPT_RELEASE_BOTH);
```

Comment:
You probably want `DB_OPT_RELEASE_DATA` here (the key is numeric, there's nothing to release). It's not safe to use options that include `DB_OPT_RELEASE_KEY` in DBMaps that have integers as keys.